source/extensions/filters/http/rate_limit_quota/quota_bucket_cache.h

```diff
@@ -41,13 +41,16 @@ struct Bucket {
   // RLQS server.
   BucketId bucket_id;
   // Cached action from the response that was received from the RLQS server.
-  BucketAction bucket_action;
+  absl::optional<BucketAction> cached_action = absl::nullopt;
+  BucketAction default_action;
```

Comment:
If default action is not necessary in this PR, i would suggest to do it in a separate PR.

Also, AFAIU when cache action expires, expired assignment behavior should be used rather than default action.